worker/transcribee_worker/whisper_transcribe.py

```diff
@@ -223,6 +223,43 @@ async def recombine_split_words(
         yield last_paragraph
 
 
+async def strict_sentence_paragraphs(
+    iter: AsyncIterator[Paragraph],
+) -> AsyncIterator[Paragraph]:
+
+    acc_paragraph = None
+    async for paragraph in iter:
+
+        # this function implements a kind of crude heuristic to seperate sentences
+        # https://www.quora.com/Do-all-living-languages-use-the-period-to-end-a-sentence
+        if paragraph.lang == "th":
+            sentence_endings = " "
+        else:
+            sentence_endings = [".", "?", "!", "。", "।", "෴", " ። ", "።", "။", ":"]
+
+        for i, atom in enumerate(paragraph.children):
+            if acc_paragraph is None:
+                acc_paragraph = Paragraph(children=[], lang=paragraph.lang)
+            acc_paragraph.children.append(atom)
+
+            # ignore something like 1. of May
+            number_dot = (
+                atom.text.endswith(".")
+                and i != 0
+                and paragraph.children[i - 1].text[-1].isnumeric
```

Comment:
isnumeric must be called, else it is always truthy. And IMO using `acc_paragraph` seems a bit cleaner, but probably doesn't change much in practice
```suggestion
                and acc_paragraph.children
                and acc_paragraph.children[-1].text.isnumeric()
```